index.js

```diff
@@ -65,12 +65,21 @@ function fastifyRedis (fastify, options, next) {
   }
 
   if (!redisOptions.lazyConnect) {
-    return client.info((err, _) => {
-      return err ? next(err) : next()
-    })
-  } else {
-    next()
+    const onError = function (err) {
+      client.quit(() => next(err))
```

Comment:
I don't think we should quit the client on the first error. In this case we are avoiding the retry strategy that is really important imho.
If we want to go with this `event` path I think we need to manually handle the retry strategy.